pkg/ocm/helpers.go

```diff
@@ -636,6 +638,49 @@ func ValidateSubnetsCount(multiAZ bool, privateLink bool, subnetsInputCount int)
 	return nil
 }
 
+func ValidateHostedClusterSubnets(awsClient aws.Client, isPrivate bool, subnetIDs []string) (int, error) {
+	if isPrivate && len(subnetIDs) < 1 {
+		return 0, fmt.Errorf("The number of subnets for a private hosted cluster should be at least one")
+	}
+	if !isPrivate && len(subnetIDs) < 2 {
+		return 0, fmt.Errorf("The number of subnets for a public hosted cluster should be at least two")
+	}
+	vpcSubnets, vpcSubnetsErr := awsClient.GetVPCSubnets(subnetIDs[0])
+	if vpcSubnetsErr != nil {
+		return 0, vpcSubnetsErr
+	}
+
+	var subnets []*ec2.Subnet
+	for _, subnet := range vpcSubnets {
+		for _, subnetId := range subnetIDs {
+			if awssdk.StringValue(subnet.SubnetId) == subnetId {
+				subnets = append(subnets, subnet)
+				break
+			}
+		}
+	}
+
+	privateSubnets, privateSubnetsErr := awsClient.FilterVPCsPrivateSubnets(subnets)
+	if privateSubnetsErr != nil {
+		return 0, privateSubnetsErr
+	}
+
+	privateSubnetCount := len(privateSubnets)
```

Comment:
consistency nit: singular "Subnet" here but plural in `privateSubnetsCount` as same value is called outside, as well as in `publicSubnetsCount`.